crowbar_framework/lib/crowbar/lock.rb

```diff
@@ -0,0 +1,76 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Crowbar
+  class Lock
+    attr_accessor :logger
+    attr_accessor :path
+    attr_accessor :remote
+
+    def initialize(options = {})
+      @logger = options.fetch :logger, Rails.logger
+      @path = options.fetch :path, "/var/lock"
+      @remote = options.fetch :remote, false
+    end
+
+    def create(name, nodes = [])
+      if @remote
+        remote(name, nodes, action: "create")
+      else
+        local(name, action: "create")
```

Comment:
No, this is definitely not the right PR to do that work, which is outside the scope of this sprint.  We need this remote locking urgently in lieu of a change we had to revert because it was bringing Jenkins to its knees, and we've already estimated that work.  So let's avoid getting distracted, get a simple solution for the remote locking working now, and have a separate story for tidying up the local locking later.

And anyway like I already said, I don't think there's any trivial way to reuse code between local and remote.  We'd need to automatically copy the library onto the other nodes, and there's an even more fundamental barrier which is that `FileLock` provides exclusive locking, whereas for this use case we need shared locking.